proto/src/determined/trial/v1/trial.proto

```diff
@@ -39,17 +39,76 @@ message TrialCheckpoint {
   // UUID of the checkpoint.
   string uuid = 1;
   // The batch number where checkpoint was evaluated.
-  int32 batch_number = 3;
+  int32 batch_number = 2;
   // Timestamp when the checkpoint began being saved to persistent storage.
-  google.protobuf.Timestamp start_time  = 4;
+  google.protobuf.Timestamp start_time  = 3;
   // Timestamp when the checkpoint completed being saved to persistent storage.
-  google.protobuf.Timestamp end_time  = 5;
+  google.protobuf.Timestamp end_time  = 4;
   // Dictionary of file paths to file sizes in bytes of all files in the checkpoint.
-  map<string, int64> resources = 6;
+  map<string, int64> resources = 5;
   // The state of the checkpoint.
-  determined.checkpoint.v1.State state = 7;
+  determined.checkpoint.v1.State state = 6;
 }
 
+// TODO ideally we can share more between these messages.
```

Comment:
maybe use these for best/latest valication and checkpoint as well

Reply:
I think we just replace `TrialCheckpoint` and `TrialValidation` with `CheckpointWorkload` and `MetricsWorkload`. how does that sound @stoksc?

Reply:
Uh, fine by me. Technically these are workload completed messages and workload is just what gets run, but that distinction probably never needs to be made to the user.